File: ainpc-core-plugin/src/main/java/ro/ainpc/world/mapping/MappingDraftKind.java
package ro.ainpc.world.mapping;

import java.util.Arrays;
import java.util.Locale;
import java.util.Optional;

public enum MappingDraftKind {
    REGION("region"),
    PLACE("place"),
    NODE("node"),
    NPC_BIND("npc_bind"),
    QUEST_ANCHOR("quest_anchor");

    private final String id;

    MappingDraftKind(String id) {
        this.id = id;
    }

    public String id() {
        return id;
    }

    public static Optional<MappingDraftKind> fromId(String value) {
        if (value == null || value.isBlank()) {
            return Optional.empty();
        }
        String normalized = value.trim().toLowerCase(Locale.ROOT).replace('-', '_');
        return Arrays.stream(values())
            .filter(kind -> kind.id.equalsIgnoreCase(normalized) || kind.name().equalsIgnoreCase(normalized))
            .findFirst();
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/world/mapping/MappingDraftKind.kt
package ro.ainpc.world.mapping

import java.util.Locale
import java.util.Optional

enum class MappingDraftKind(private val idValue: String) {
    REGION("region"),
    PLACE("place"),
    NODE("node"),
    NPC_BIND("npc_bind"),
    QUEST_ANCHOR("quest_anchor");

    fun id(): String = idValue

    companion object {
        @JvmStatic
        fun fromId(value: String?): Optional<MappingDraftKind> {
            if (value.isNullOrBlank()) {
                return Optional.empty()
            }
            val normalized = value.trim()
                .lowercase(Locale.ROOT)
                .replace('-', '_')
            for (kind in values()) {
                if (kind.idValue.equals(normalized, ignoreCase = true) ||
                    kind.name.equals(normalized, ignoreCase = true)
                ) {
                    return Optional.of(kind)
                }
            }
            return Optional.empty()
        }
    }
}
